tests/jobserver/views/test_projects.py

```diff
@@ -283,11 +284,57 @@ def test_projectsettings_success(rf, superuser):
     assert response.context_data["project"] == project
 
 
+@pytest.mark.django_db
+def test_projectsettings_post_failure(rf, superuser):
+    org = OrgFactory()
+    project = ProjectFactory(org=org)
+
+    ProjectInvitationFactory(project=project)
+    assert ProjectInvitation.objects.filter(project=project).count() == 1
+
+    request = rf.post(MEANINGLESS_URL, {"users": ["not_a_pk"]})
+    request.user = superuser
+
+    response = ProjectSettings.as_view()(
+        request, org_slug=org.slug, project_slug=project.slug
+    )
+
+    assert response.status_code == 200
+
+    # check the number of invitations hasn't changed
+    assert ProjectInvitation.objects.filter(project=project).count() == 1
+
+    assert "“not_a_pk” is not a valid value." in response.rendered_content
+
+
+@pytest.mark.django_db
+def test_projectsettings_post_success(rf, superuser):
+    org = OrgFactory()
+    project = ProjectFactory(org=org)
+    invitee = UserFactory()
+    user = UserFactory()
+
+    ProjectInvitationFactory(project=project, user=invitee)
+    assert ProjectInvitation.objects.filter(project=project).count() == 1
+
+    request = rf.post(MEANINGLESS_URL, {"users": [str(user.pk)]})
+    request.user = superuser
+
+    response = ProjectSettings.as_view()(
+        request, org_slug=org.slug, project_slug=project.slug
+    )
+
+    assert response.status_code == 302
+    assert response.url == project.get_settings_url()
+
+    assert ProjectInvitation.objects.filter(project=project).count() == 2
```

Comment:
You might like to pinch the [`assert_difference` function from opencodelists](https://github.com/opensafely-core/opencodelists/blob/master/opencodelists/tests/assertions.py).